pontis-package/pontis/core/base_client.py

```diff
@@ -109,7 +111,16 @@ async def send_transactions(self, calls, max_fee=None):
         calldata = [x for call in calls for x in call[2]]
 
         # Estimate fee
-        prepared = self.account_contract.functions["__execute__"].prepare(
+        with open(path.join(path.dirname(__file__), "../abi/Account.json"), "r") as f:
+            account_abi = json.load(f)
+        contract_data = ContractData.from_abi(
+            self.account_contract_address, account_abi
+        )
+        execute_abi = [a for a in account_abi if a["name"] == "__execute__"][0]
```

Comment:
Otherwise code looks good I just need to understand this section better. Reminds me a lot of writing Terra multicalls in rust or even their terra.js lib (rip Terra lol)

Reply:
Good question! Read this on list comprehension in Python: https://www.w3schools.com/python/python_lists_comprehension.asp
What I am doing here is iterating through `account_abi`, which is a list of dicts and selecting the the dict whose "name" entry is "__execute__", i.e. the ABI for the execute call